python/lsst/obs/base/butler_tests.py

```diff
@@ -0,0 +1,186 @@
+                  'sky_origin',
+                  'raw_subsets',
+                  'good_ccds',
+                  'bad_ccds',
+                  'linearizer'
+                  ]
+        ButlerGet = collections.namedtuple("ButlerGetData", fields)
+
+        self.butlerGetData = ButlerGet(ccdExposureId_bits=ccdExposureId_bits,
+                                       exposureIds=exposureIds,
+                                       filters=filters,
+                                       exptimes=exptimes,
+                                       detectorIds=detectorIds,
+                                       detector_names=detector_names,
+                                       detector_serials=detector_serials,
+                                       dimensions=dimensions,
+                                       sky_origin=sky_origin,
+                                       raw_subsets=raw_subsets,
+                                       good_ccds=good_ccds,
+                                       bad_ccds=bad_ccds,
+                                       linearizer=linearizer
+                                       )
+
+    def tearDown(self):
+        del self.butler
+        super(ButlerGetTests, self).tearDown()
+
+    def test_id(self):
+        bits = self.butler.get('ccdExposureId_bits')
+        self.assertEqual(bits, self.butlerGetData.ccdExposureId_bits)
+        exposureId = self.butler.get('ccdExposureId', dataId=self.dataIds['raw'])
+        self.assertEqual(exposureId, self.butlerGetData.exposureIds['raw'])
+
+    def _test_exposure(self, name, hasWcs=False):
+        if self.dataIds[name] is unittest.SkipTest:
+            self.skipTest('Skipping %s as requested'%(inspect.currentframe().f_code.co_name))
+        exp = self.butler.get(name, self.dataIds[name])
+        self.assertEqual(exp.getDimensions(), self.butlerGetData.dimensions[name])
+        self.assertEqual(exp.getDetector().getId(), self.butlerGetData.detectorIds[name])
+        self.assertEqual(exp.getDetector().getName(), self.butlerGetData.detector_names[name])
+        self.assertEqual(exp.getDetector().getSerial(), self.butlerGetData.detector_serials[name])
+        self.assertEqual(exp.getFilter().getName(), self.butlerGetData.filters[name])
+        self.assertEqual(exp.hasWcs(), hasWcs)
+        exposureId = self.butler.get('ccdExposureId', dataId=self.dataIds[name])
+        self.assertEqual(exposureId, self.butlerGetData.exposureIds[name])
+        self.assertEqual(exp.getInfo().getVisitInfo().getExposureTime(), self.butlerGetData.exptimes[name])
+        return exp
+
+    def test_raw(self):
+        exp = self._test_exposure('raw', hasWcs=True)
+        # additional tests of the wcs of a raw exposure
+        origin = exp.getWcs().getSkyOrigin()
+        self.assertAlmostEqual(origin.getLongitude().asDegrees(), self.butlerGetData.sky_origin[0])
+        self.assertAlmostEqual(origin.getLatitude().asDegrees(), self.butlerGetData.sky_origin[1])
```

Comment:
Can I suggest also testing the ability to get sub-images?

    sub = self.butler.get("raw_sub", dataId, bbox=afwGeom.Box2I(...))